core/src/main/java/org/infinispan/jmx/AbstractJmxRegistration.java

```diff
@@ -73,26 +80,68 @@ public void start() {
          }
 
          if (mBeanServer != null) {
-            jmxDomain = initDomain(mBeanServer);
+            // first time!
+            groupName = initGroup();
+
+            List<ResourceDMBean> mbeans = getResourceDMBeansFromComponents();
+            Iterator<ResourceDMBean> it = mbeans.iterator();
+            ResourceDMBean first = it.next();
+
+            GlobalJmxStatisticsConfiguration globalJmxConfig = globalConfig.globalJmxStatistics();
+            String jmxDomain = globalJmxConfig.domain();
+            int counter = 2;
+            while (true) {
```

Comment:
I suggest extracting the while loop in a separate method

Reply:
I'd like to keep them together so they can die together. The registration of first bean and the rest and the loopy loop will disappear in spring next year when 11 removes allowDuplicateDomains. I've added the jira link there.